src/imap_client.rs

```diff
@@ -38,7 +38,7 @@ impl Client {
         let stream = TcpStream::connect(addr).await?;
         let tls_config = dc_build_tls_config(certificate_checks);
         let tls_connector: async_tls::TlsConnector = Arc::new(tls_config).into();
-        let tls_stream = tls_connector.connect(domain.as_ref(), stream)?.await?;
+        let tls_stream = tls_connector.connect(domain.as_ref(), stream).await?;
```

Comment:
is this change intentionally?

Reply:
yes -- it's a change actually unrelated to what i tried to do here, just from upgrading tls